File: 1603.design-parking-system/solution.kt
class ParkingSystem(big: Int, medium: Int, small: Int) {
    val capacities = intArrayOf(big, medium, small)

    fun addCar(carType: Int): Boolean {
        this.capacities[carType - 1] -= 1
        return this.capacities[carType - 1] >= 0
    }

}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * var obj = ParkingSystem(big, medium, small)
 * var param_1 = obj.addCar(carType)
 */

File: 1603.design-parking-system/solution.py
class ParkingSystem:

    def __init__(self, big: int, medium: int, small: int):
        self.capacities = [big, medium, small]
        self.status = [0, 0, 0]

    def addCar(self, carType: int) -> bool:
        self.status[carType - 1] += 1
        return (self.status[carType - 1] <= self.capacities[carType - 1])


# Your ParkingSystem object will be instantiated and called as such:
# obj = ParkingSystem(big, medium, small)
# param_1 = obj.addCar(carType)
